src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -343,8 +326,13 @@ void ObjectMonitor::enter_for_with_contention_mark(JavaThread* locking_thread, O
     // Racing with deflation.
     prev_owner = try_set_owner_from(DEFLATER_MARKER, locking_thread);
     if (prev_owner == DEFLATER_MARKER) {
-      // Cancelled deflation. Increment contentions as part of the deflation protocol.
-      add_to_contentions(1);
+      // We successfully cancelled the in-progress async deflation.
+      // By extending the lifetime of the contention_mark, we
+      // prevent the destructor from decrementing the contentions
+      // counter when the contention_mark goes out of scope. Instead
+      // ObjectMonitor::deflate_monitor() will decrement contentions
+      // after it recognizes that the async deflation was cancelled.
+      contention_mark.extend();
```

Comment:
I think extends() should add to contentions in this PR since extends() is part of this PR, and initially I expected it to be sort of a refcount (with a case or two for extending the scope of the refcount).